hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -832,17 +832,22 @@ private void connectToClusterInternal() {
                 throw new IllegalStateException("Client is being shutdown.");
             }
 
-            final long remainingTime = nextTry - Clock.currentTimeMillis();
-            logger.warning(String.format("Unable to get alive cluster connection, try in %d ms later, attempt %d of %d.",
-                    Math.max(0, remainingTime), attempt, connectionAttemptLimit));
-
-            if (remainingTime > 0) {
-                try {
-                    Thread.sleep(remainingTime);
-                } catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    break;
+            if (attempt < connectionAttemptLimit) {
```

Comment:
Sleep is not related to consuming attempts. `connectionAttemptPeriod`  is meant to be a constant period whether all connections are tried or not.